pkg/oc/bootstrap/clusterup/components/persistent-volumes/setup_persistent_volumes.go

```diff
@@ -0,0 +1,250 @@
+package persistent_volumes
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift/origin/pkg/oc/admin/policy"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/serviceaccount"
+	kbatch "k8s.io/kubernetes/pkg/apis/batch"
+	kapi "k8s.io/kubernetes/pkg/apis/core"
+	kubernetes "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/staging/src/k8s.io/apimachinery/pkg/util/wait"
+
+	authorizationtypedclient "github.com/openshift/origin/pkg/authorization/generated/internalclientset/typed/authorization/internalversion"
+	"github.com/openshift/origin/pkg/oc/bootstrap/clusterup/componentinstall"
+	"github.com/openshift/origin/pkg/oc/bootstrap/docker/dockerhelper"
+	securityclient "github.com/openshift/origin/pkg/security/generated/internalclientset"
+	securitytypedclient "github.com/openshift/origin/pkg/security/generated/internalclientset/typed/security/internalversion"
+)
+
+const (
+	pvCount            = 100
+	pvSetupJobName     = "persistent-volume-setup"
+	pvInstallerSA      = "pvinstaller"
+	pvTargetNamespace  = "default"
+	pvIgnoreMarkerFile = ".skip_pv"
+)
+
+// TODO: This really should be a Go template
```

Comment:
How do you want to make this all a go template, I'm curious.

Reply:
move it to its own file, parametrize the template instead of just doing Sprintf(), to me go template sounds like a better fit for this